get_event_attendees error message and documentation + README for past events

RickPack opened the pull request:
For Issue #22  @ledell 
Internals.R: Custom get_event_attendees error message to indicate user must be a group administration to use htis function due to Meetup API restriction. Updated .R and README documentation for function. 
Also README updated to show past events because query returned 0 records this weekend (no planned events, which is the default argument).

The change to the README authors in the MD file appears to be because it had previously not been knit after the addition of MIT, and I added a comma before that interest in the author sequence. The single-word usage of "TODO" appeared to have also not been previously knit.

RickPack committed:
new error message when group admin status needed (get_event_attendees), README queries past events to avoid 0 records returning

RickPack committed:
Restored TO DO separation

RickPack committed:
extraneous code removed

RickPack committed:
TODO as one word to match README

RickPack committed:
README comma and had forgotten to knit

ledell commented on R/get_event_attendees.R:
```diff
@@ -23,9 +24,9 @@
-                    "/events/",
-                    event_id,
-                    "/attendance")
+                       "/events/",
```
Can you remove these extra tabs?  Should not be there.

ledell commented on R/internals.R:
```diff
@@ -14,7 +14,12 @@
   req <- httr::GET(url = api_url,          # the endpoint
                    query = parameters)
-
+  if(req$status_code == 400){
```
Please using spacing according to [Hadley's style guide](http://r-pkgs.had.co.nz/style.html): 
```
if (req$status_code == 400) {
```
and can you change the message to: "HTTP 400 error encountered.  Note for some endpoints the Meetup API requires you to be the respective group’s administrator."

LucyMcGowan replied:
Can you change this message slightly to something like: 
"HTTP 400 error encountered. Note for some endpoints the Meetup API requires you to be the respective group’s administrator."

ledell commented on README.Rmd:
```diff
@@ -19,7 +19,7 @@ knitr::opts_chunk$set(
 R interface to the Meetup API (v3)
 
-**Authors:** [Gabriela de Queiroz](http://gdequeiroz.github.io/), [Erin LeDell](http://www.stat.berkeley.edu/~ledell/), [Olga Mierzwa-Sulima](https://github.com/olgamie), [Lucy D'Agostino McGowan](http://www.lucymcgowan.com), [Claudia Vitolo](https://github.com/cvitolo)<br/> [MIT](https://opensource.org/licenses/MIT)
+**Authors:** [Gabriela de Queiroz](http://gdequeiroz.github.io/), [Erin LeDell](http://www.stat.berkeley.edu/~ledell/), [Olga Mierzwa-Sulima](https://github.com/olgamie), [Lucy D'Agostino McGowan](http://www.lucymcgowan.com), [Claudia Vitolo](https://github.com/cvitolo)<br/>, [MIT](https://opensource.org/licenses/MIT)
```
That comma and extra link to "MIT" should not be there.  Do you think you can remove this? `, [MIT](https://opensource.org/licenses/MIT)`  That might have been a typo on rladies:master, and it looks like you tried to fix it with a comma, but we can just remove that.  Thanks!

ledell commented on README.md:
```diff
@@ -7,18 +7,18 @@ meetupr
 R interface to the Meetup API (v3)
 
-**Authors:** [Gabriela de Queiroz](http://gdequeiroz.github.io/), [Erin LeDell](http://www.stat.berkeley.edu/~ledell/), [Olga Mierzwa-Sulima](https://github.com/olgamie), [Lucy D'Agostino McGowan](http://www.lucymcgowan.com), [Claudia Vitolo](https://github.com/cvitolo)<br/> **License:** [MIT](https://opensource.org/licenses/MIT)
+**Authors:** [Gabriela de Queiroz](http://gdequeiroz.github.io/), [Erin LeDell](http://www.stat.berkeley.edu/~ledell/), [Olga Mierzwa-Sulima](https://github.com/olgamie), [Lucy D'Agostino McGowan](http://www.lucymcgowan.com), [Claudia Vitolo](https://github.com/cvitolo)<br/>, [MIT](https://opensource.org/licenses/MIT) **License:** [MIT](https://opensource.org/licenses/MIT)
```
Same comment as above, comma and extra link to "MIT" should not be there.

ledell commented on man/get_boards.Rd:
```diff
@@ -7,7 +7,7 @@
 }
 \arguments{
-\item{urlname}{Character. The name of the group as indicicated in the
+\item{urlname}{Character. The name of the group as indicated in the
```
This change was already fixed on master.  Same goes for all Rd files below.

ledell replied:
Ah... nevermind. When I fixed the typo, I forgot to re-generate all the Rd files, which is why they are all showing up as changed here.  Thanks!

LucyMcGowan commented on R/get_event_attendees.R:
```diff
@@ -1,5 +1,6 @@
 #' Get the attendees for a specified event
 #'
+#' Must be an administrator of the queried Meetup group because of Meetup.com API restriction on get_event_attendees
```
Can you link to this: https://www.meetup.com/meetup_api/docs/:urlname/events/:id/attendance/#list
maybe like:

Must be an administrator of the queried Meetup group due to the [Meetup.com API restrictions](https://www.meetup.com/meetup_api/docs/:urlname/events/:id/attendance/#list) on this endpoint.

LucyMcGowan reviewed (changes requested):
This looks great! I just have a few small change requests. Thank you @RickPack 🎉

ledell commented:
@RickPack @LucyMcGowan Ok, this is a bit weird... I checked out your fork, and I am testing the `get_event_attendees()` function using the demo (Nashville R-Ladies meetup event), and it's letting me get the results, even though I am not an admin of that group.  I thought that was supposed to trigger the error?  

```
> urlname <- "rladies-nashville"
> past_events <- get_events(urlname = urlname, api_key = api_key,
+                           event_status = "past")
Downloading 9 record(s)...
> event_id <- past_events$id[1]  #first event for this group
> attendees <- get_event_attendees(urlname, event_id)
Downloading 27 record(s)...
> attendees
# A tibble: 27 x 4
          id                     name   status   resource
       <int>                    <chr>    <chr>     <list>
 1 126309962            Aliya Gifford attended <list [3]>
 2 211530139         Allison Hainline attended <list [3]>
 3 215531726                  Chris S attended <list [3]>
 4 214589711             Dilman Yasin attended <list [3]>
 5 215533912       Elizabeth Sigworth attended <list [3]>
 6 215531478        Fatemeh Dalilian  attended <list [3]>
 7 213947104                   Hannah attended <list [3]>
 8 211111732             Hannah Weeks attended <list [3]>
 9 214763447          Jackie Pennings attended <list [3]>
10 201661720 Jacquelyn Elizabeth Neal attended <list [3]>
# ... with 17 more rows
```

LucyMcGowan commented:
@ledell: I think the difference is @RickPack added the `event_status = "past"` which makes the information accessible to non-admins (if it was left to the default, I believe it wouldn't work).

LucyMcGowan commented:
☝️ That is also making me realize some of my attendees full names are now printed in our README, I'm not sure that anyone will mind, but maybe I should check? Or maybe we don't print that part?

ledell commented:
@LucyMcGowan @RickPack Oh, ok...  I thought that all attendee data was not accessible to non-admins.  So this makes sense, now, thanks!

In that case, @RickPack let's just remove the `*Note: Must be a Meetup group administrator to use because of API restriction.` in the README, and put the details in the documentation for the `get_event_attendees()` function instead.  Thx.

ledell commented:
@LucyMcGowan 
>  That is also making me realize some of my attendees full names are now printed in our README, I'm not sure that anyone will mind, but maybe I should check? Or maybe we don't print that part?

That's up to you, but yeah, probably best to remove the print-out for privacy-ish reasons. cc @RickPack

RickPack committed:
removed names from README, get_event_attendees no longer has admin restricton, clean-up error message for HTTP 400

RickPack commented:
Well friends, I made some changes and then I reverted some changes because the Meetup.com API no longer has the restriction on querying attendees. This may have changed just today. For example, this line that @ledell and I reported in #27 as returning HTTP 400 now works:
`get_event_attendees(urlname,` event_id = 234968855)`

I believe I addressed everything else. Please feel free to make whatever updates you like (if that's possible). Work performed included: removed name printing from README, get_event_attendees no longer has admin restriction so removed documentation, spacing in internals.R, clean-up error message for HTTP 400 (in case it ever occurs now or in the future)

I did not see the extra spaces in `get_event_attendees.R`. It looks like the rladies version on my screen. Could be something odd about my setup.

NOTE: Unknown checks failed a moment ago. They are now pending so I'll review in the morning.

RickPack committed:
README knit and internals.R spacing

RickPack committed:
Attempting to eliminate non-ascii character build error in internals.R. Addition of period should not work...

RickPack committed:
Violates code standard but long single-line attempt in internals.R to eliminate non-ascii character error in Github

RickPack committed:
Still attempting to eliminate non-ASCII Github build error.

RickPack commented:
Github Travis CI build reports a non-ascii character in internals.R. I have tried solutions including a search for regular expression  [^\x00-\x7F]+ per [StackOverflow](https://stackoverflow.com/questions/20889996/notepad-how-to-remove-all-non-ascii-characters-with-regex) and setting the RStudio encoding explicitly to ASCII and then saving. I will see if a nominal edit somehow causes the error to disappear and if that fails, perhaps my internals.R update can be discarded (just added a stop message for HTTP 400) and someone can try adding the lines themselves.

BTW, thank you for your help with this pull request. I have learned a great deal.

RickPack commented:
I generated the HTTP 400 with this line:
`meetupr::get_event_attendees("Research-Triangle-Analysts", 246678392)`
but it does not appear with this line:
`get_event_attendees("rladies-nashville", event_id = 234968855)`

I am not an rladies-nashville administrator.
This will take a bit of research. I may not get to this until Sunday, March 11, 2018.

ledell commented:
Hi @RickPack do you know if this PR is still needed/useful?  A lot has changed since then.  I am trying to clean up & merge all outstanding PRs.  Sorry that this didn't get merged before.

maelle commented:
@ledell should this be closed?

maelle commented:
in which case you should also close #22
